About the dataset - [Kaggle](https://www.kaggle.com/datasets/airbnb/seattle)

**Context**

Since 2008, guests and hosts have used Airbnb to travel in a more unique, personalized way. As part of the Airbnb Inside initiative, this dataset describes the listing activity of homestays in Seattle, WA.

**Content**

The following Airbnb activity is included in this Seattle dataset:

Listings, including full descriptions and average review score
Reviews, including unique id for each reviewer and detailed comments
Calendar, including listing id and the price and availability for that day

#### Data Cleaning using Python

In [ ]:
import pandas as pd

In [ ]:
# Store CSV to DataFrame
listing_host_csv = "/content/drive/MyDrive/DS Portfolio/Projects and Case Studies/AirBnB/listings.csv"
listing_host_df = pd.read_csv(listing_host_csv,encoding="utf8")

In [ ]:
# change column "name" to something that SQL doesn't already recognize
listing_host_df.rename(index=str,columns={"name":"listing_name"},inplace=True)
listing_host_df.head()

id                           listing_url       scrape_id last_scraped  \
0   241032   https://www.airbnb.com/rooms/241032  20160104002432   2016-01-04   
1   953595   https://www.airbnb.com/rooms/953595  20160104002432   2016-01-04   
2  3308979  https://www.airbnb.com/rooms/3308979  20160104002432   2016-01-04   
3  7421966  https://www.airbnb.com/rooms/7421966  20160104002432   2016-01-04   
4   278830   https://www.airbnb.com/rooms/278830  20160104002432   2016-01-04   

                          listing_name  \
0         Stylish Queen Anne Apartment   
1   Bright & Airy Queen Anne Apartment   
2  New Modern House-Amazing water view   
3                   Queen Anne Chateau   
4       Charming craftsman 3 bdm house   

                                             summary  \
0                                                NaN   
1  Chemically sensitive? We've removed the irrita...   
2  New modern house built in 2013.  Spectacular s...   
3  A charming apartment that sits atop Queen Anne...   
4  Cozy family craftman house in beautiful neighb...   

                                               space  \
0  Make your self at home in this charming one-be...   
1  Beautiful, hypoallergenic apartment in an extr...   
2  Our house is modern, light and fresh with a wa...   
3                                                NaN   
4  Cozy family craftman house in beautiful neighb...   

                                         description experiences_offered  \
0  Make your self at home in this charming one-be...                none   
1  Chemically sensitive? We've removed the irrita...                none   
2  New modern house built in 2013.  Spectacular s...                none   
3  A charming apartment that sits atop Queen Anne...                none   
4  Cozy family craftman house in beautiful neighb...                none   

                               neighborhood_overview  ... review_scores_value  \
0                                                NaN  ...                10.0   
1  Queen Anne is a wonderful, truly functional vi...  ...                10.0   
2  Upper Queen Anne is a charming neighborhood fu...  ...                10.0   
3                                                NaN  ...                 NaN   
4  We are in the beautiful neighborhood of Queen ...  ...                 9.0   

  requires_license license jurisdiction_names instant_bookable  \
0                f     NaN         WASHINGTON                f   
1                f     NaN         WASHINGTON                f   
2                f     NaN         WASHINGTON                f   
3                f     NaN         WASHINGTON                f   
4                f     NaN         WASHINGTON                f   

  cancellation_policy  require_guest_profile_picture  \
0            moderate                              f   
1              strict                              t   
2              strict                              f   
3            flexible                              f   
4              strict                              f   

  require_guest_phone_verification calculated_host_listings_count  \
0                                f                              2   
1                                t                              6   
2                                f                              2   
3                                f                              1   
4                                f                              1   

  reviews_per_month  
0              4.07  
1              1.48  
2              1.15  
3               NaN  
4              0.89  

[5 rows x 92 columns]

In [ ]:
# Create new data with select columns for the AirBnB listings
listing_df = listing_host_df[["id","listing_name","street","neighbourhood_cleansed","neighbourhood_group_cleansed","city","state","zipcode","latitude","longitude","is_location_exact","property_type","room_type","accommodates","bathrooms","bedrooms","beds","bed_type","square_feet","price","weekly_price","monthly_price","security_deposit","cleaning_fee","guests_included","extra_people","minimum_nights","maximum_nights","has_availability","availability_30","availability_60","availability_90","availability_365","number_of_reviews","first_review","last_review","review_scores_rating","review_scores_accuracy","review_scores_cleanliness","review_scores_checkin","review_scores_communication","review_scores_location","review_scores_value","requires_license","instant_bookable","cancellation_policy","require_guest_profile_picture","require_guest_phone_verification","reviews_per_month","host_id"]].copy()

In [ ]:
# convert dates to datetime
listing_df["first_review"]=pd.to_datetime(listing_df["first_review"])
listing_df["last_review"]=pd.to_datetime(listing_df["last_review"])

In [ ]:
# convert boolean columns to boolean
listing_df["is_location_exact"].replace(["t","f"],[True,False],inplace=True)
listing_df["has_availability"].replace(["t","f"],[True,False],inplace=True)
listing_df["requires_license"].replace(["t","f"],[True,False],inplace=True)
listing_df["instant_bookable"].replace(["t","f"],[True,False],inplace=True)
listing_df["require_guest_profile_picture"].replace(["t","f"],[True,False],inplace=True)
listing_df["require_guest_phone_verification"].replace(["t","f"],[True,False],inplace=True)

In [ ]:
# convert all the currency columns to numeric values instead of strings
# define the currency columns
currency_cols=["price","weekly_price","monthly_price","security_deposit","cleaning_fee","extra_people"]
# remove dollar sign and commas
listing_df[currency_cols]=listing_df[currency_cols].replace({'\$': '', ',': ''}, regex=True)
# changing the leftover value (minus currency sign and commas) to a number
listing_df["price"] = pd.to_numeric(listing_df["price"])
listing_df["weekly_price"] = pd.to_numeric(listing_df["weekly_price"])
listing_df["monthly_price"] = pd.to_numeric(listing_df["monthly_price"])
listing_df["security_deposit"] = pd.to_numeric(listing_df["security_deposit"])
listing_df["cleaning_fee"] = pd.to_numeric(listing_df["cleaning_fee"])
listing_df["extra_people"] = pd.to_numeric(listing_df["extra_people"])
listing_df

id                         listing_name  \
0       241032         Stylish Queen Anne Apartment   
1       953595   Bright & Airy Queen Anne Apartment   
2      3308979  New Modern House-Amazing water view   
3      7421966                   Queen Anne Chateau   
4       278830       Charming craftsman 3 bdm house   
...        ...                                  ...   
3813   8101950   3BR Mountain View House in Seattle   
3814   8902327    Portage Bay View!-One Bedroom Apt   
3815  10267360    Private apartment view of Lake WA   
3816   9604740    Amazing View with Modern Comfort!   
3817  10208623            Large Lakefront Apartment   

                                                 street  \
0         Gilman Dr W, Seattle, WA 98119, United States   
1     7th Avenue West, Seattle, WA 98119, United States   
2     West Lee Street, Seattle, WA 98119, United States   
3     8th Avenue West, Seattle, WA 98119, United States   
4          14th Ave W, Seattle, WA 98119, United States   
...                                                 ...   
3813  Northwest 48th Street, Seattle, WA 98107, Unit...   
3814  Fuhrman Avenue East, Seattle, WA 98102, United...   
3815  South Laurel Street, Seattle, WA 98178, United...   
3816  43rd Avenue East, Seattle, WA 98112, United St...   
3817  Westlake Avenue North, Seattle, WA 98109, Unit...   

     neighbourhood_cleansed neighbourhood_group_cleansed     city state  \
0           West Queen Anne                   Queen Anne  Seattle    WA   
1           West Queen Anne                   Queen Anne  Seattle    WA   
2           West Queen Anne                   Queen Anne  Seattle    WA   
3           West Queen Anne                   Queen Anne  Seattle    WA   
4           West Queen Anne                   Queen Anne  Seattle    WA   
...                     ...                          ...      ...   ...   
3813                Fremont          Other neighborhoods  Seattle    WA   
3814            Portage Bay                 Capitol Hill  Seattle    WA   
3815          Rainier Beach               Rainier Valley  Seattle    WA   
3816           Madison Park                 Capitol Hill  Seattle    WA   
3817        East Queen Anne                   Queen Anne  Seattle    WA   

     zipcode   latitude   longitude  ...  review_scores_communication  \
0      98119  47.636289 -122.371025  ...                         10.0   
1      98119  47.639123 -122.365666  ...                         10.0   
2      98119  47.629724 -122.369483  ...                         10.0   
3      98119  47.638473 -122.369279  ...                          NaN   
4      98119  47.632918 -122.372471  ...                         10.0   
...      ...        ...         ...  ...                          ...   
3813   98107  47.664295 -122.359170  ...                          8.0   
3814   98102  47.649552 -122.318309  ...                         10.0   
3815   98178  47.508453 -122.240607  ...                          NaN   
3816   98112  47.632335 -122.275530  ...                          NaN   
3817   98109  47.641186 -122.342085  ...                          NaN   

     review_scores_location review_scores_value  requires_license  \
0                       9.0                10.0             False   
1                      10.0                10.0             False   
2                      10.0                10.0             False   
3                       NaN                 NaN             False   
4                       9.0                 9.0             False   
...                     ...                 ...               ...   
3813                   10.0                 8.0             False   
3814                   10.0                10.0             False   
3815                    NaN                 NaN             False   
3816                    NaN                 NaN             False   
3817                    NaN                 NaN             False   

      instant_bookable  cancellatio

In [ ]:
# Create new data with select columns for the AirBnB hosts
host_df = listing_host_df[["host_id","host_name","host_since","host_location","host_response_time","host_response_rate","host_acceptance_rate","host_is_superhost","host_neighbourhood","host_listings_count","host_has_profile_pic","host_identity_verified"]].copy()

In [ ]:
# delete duplicates
host_df.drop_duplicates(keep="first",inplace=True)

In [ ]:
# convert dates to datetime format
host_df["host_since"]=pd.to_datetime(host_df["host_since"])

In [ ]:
# convert boolean columns to boolean
host_df["host_is_superhost"].replace(["t","f"],[True,False],inplace=True)
host_df["host_has_profile_pic"].replace(["t","f"],[True,False],inplace=True)
host_df["host_identity_verified"].replace(["t","f"],[True,False],inplace=True)

In [ ]:
# converting the percentage columns from strings to percentages
# remove the percentage symbols from the columns with percentages
host_df["host_response_rate"]=host_df["host_response_rate"].replace({'\$': '', ',': '', '%':''}, regex=True)
host_df["host_acceptance_rate"]=host_df["host_acceptance_rate"].replace({'\$': '', ',': '', '%':''}, regex=True)
# convert the percentage columns to numbers
host_df["host_response_rate"] = pd.to_numeric(host_df["host_response_rate"])
host_df["host_acceptance_rate"] = pd.to_numeric(host_df["host_acceptance_rate"])
host_df

host_id host_name host_since                       host_location  \
0       956883     Maija 2011-08-11  Seattle, Washington, United States   
1      5177328    Andrea 2013-02-21  Seattle, Washington, United States   
2     16708587      Jill 2014-06-12  Seattle, Washington, United States   
3      9851441     Emily 2013-11-06  Seattle, Washington, United States   
4      1452570     Emily 2011-11-29  Seattle, Washington, United States   
...        ...       ...        ...                                 ...   
3811  11847918      Ryan 2014-02-01  Seattle, Washington, United States   
3814  46566046      Glen 2015-10-14  Seattle, Washington, United States   
3815  52791370  Virginia 2015-12-30                                  US   
3816  25522052     Karen 2015-01-03   Tacoma, Washington, United States   
3817  14703116       Gil 2014-04-25  Seattle, Washington, United States   

      host_response_time  host_response_rate  host_acceptance_rate  \
0     within a few hours                96.0                 100.0   
1         within an hour                98.0                 100.0   
2     within a few hours                67.0                 100.0   
3                    NaN                 NaN                   NaN   
4         within an hour               100.0                   NaN   
...                  ...                 ...                   ...   
3811  within a few hours               100.0                 100.0   
3814      within an hour               100.0                 100.0   
3815                 NaN                 NaN                   NaN   
3816      within an hour               100.0                   NaN   
3817        within a day               100.0                   NaN   

     host_is_superhost host_neighbourhood  host_listings_count  \
0                False         Queen Anne                  3.0   
1                 True         Queen Anne                  6.0   
2                False         Queen Anne                  2.0   
3                False         Queen Anne                  1.0   
4                False         Queen Anne                  2.0   
...                ...                ...                  ...   
3811             False            Fremont                  1.0   
3814             False        Portage Bay                  1.0   
3815             False                NaN                  1.0   
3816             False                NaN                  1.0   
3817             False         Queen Anne                  1.0   

     host_has_profile_pic host_identity_verified  
0                    True                   True  
1                    True                   True  
2                    True                   True  
3                    True                   True  
4                    True                   True  
...                   ...                    ...  
3811                 True                   True  
3814                 True                   True  
3815                 True                  False  
3816                 True                   True  
3817                 True                   True  

[2751 rows x 12 columns]

In [ ]:
# Store CSV to DataFrame
availability_file = "/content/drive/MyDrive/DS Portfolio/Projects and Case Studies/AirBnB/calendar.csv"
availability_df = pd.read_csv(availability_file,encoding="utf8")

In [ ]:
# change column "date" to something that SQL doesn't already recognize
availability_df.rename(index=str,columns={"date":"available_date"},inplace=True)
# convert column to datetime
availability_df["available_date"]=pd.to_datetime(availability_df["available_date"])

In [ ]:
# convert boolean columns to boolean
availability_df["available"].replace(["t","f"], [True,False], inplace=True)

In [ ]:
# remove dollar sign
availability_df["price"] = availability_df["price"].replace({'\$': '', ',': ''}, regex=True)
# convert the price column to numeric values instead of strings
availability_df["price"] = pd.to_numeric(availability_df["price"])
availability_df.head()

listing_id available_date  available  price
0      241032     2016-01-04       True   85.0
1      241032     2016-01-05       True   85.0
2      241032     2016-01-06      False    NaN
3      241032     2016-01-07      False    NaN
4      241032     2016-01-08      False    NaN

In [ ]:
# Store CSV to DataFrame
reviews_file = "/content/drive/MyDrive/DS Portfolio/Projects and Case Studies/AirBnB/reviews.csv"
reviews_df = pd.read_csv(reviews_file,encoding="utf8")

In [ ]:
# change column "date/id" to something that SQL doesn't already recognize
reviews_df.rename(index=str,columns={"date":"review_date"},inplace=True)
reviews_df.rename(columns={"id":"review_id"}, inplace=True)

In [ ]:
# Create new data with select columns for the AirBnB listings
review_df = reviews_df[['review_id', "listing_id", "review_date", "reviewer_id", "reviewer_name", "comments"]].copy()

In [ ]:
# convert dates to datetime
review_df["review_date"] = pd.to_datetime(review_df["review_date"])
review_df.head()

review_id  listing_id review_date  reviewer_id reviewer_name  \
0   38917982     7202016  2015-07-19     28943674        Bianca   
1   39087409     7202016  2015-07-20     32440555         Frank   
2   39820030     7202016  2015-07-26     37722850           Ian   
3   40813543     7202016  2015-08-02     33671805        George   
4   41986501     7202016  2015-08-10     34959538          Ming   

                                            comments  
0  Cute and cozy place. Perfect location to every...  
1  Kelly has a great room in a very central locat...  
2  Very spacious apartment, and in a great neighb...  
3  Close to Seattle Center and all it has to offe...  
4  Kelly was a great host and very accommodating ...

#### SQL

In [ ]:
import sqlite3

In [ ]:
conn = sqlite3.connect('airbnb.db')
cursor = conn.cursor()


In [ ]:
# Create the listing table
listing_df.to_sql('listing', conn, if_exists='replace', index=False)

# Create the host table
host_df.to_sql('host', conn, if_exists='replace', index=False)

# Create the review table
review_df.to_sql('review', conn, if_exists='replace', index=False)

# Create the availability table
availability_df.to_sql('availability', conn, if_exists='replace', index=False)



1393570

#### Exploring the differences between superhosts and other hosts across a variety of metrics.

Price of listings: Do superhosts or other hosts have more expensive listings?

In [ ]:
query = """
    SELECT
        h.host_is_superhost,
        AVG(l.price) AS average_price
    FROM
        host AS h
    JOIN
        listing AS l ON h.host_id = l.host_id
    GROUP BY
        h.host_is_superhost
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
for row in results:
    host_type = "Superhost" if row[0] == 1 else "Other Host"
    average_price = row[1]
    print(f"{host_type}: Average Price - ${average_price:.2f}")


Other Host: Average Price - $174.50
Other Host: Average Price - $127.39
Superhost: Average Price - $130.14


Based on the output, the average price of listings for Other Hosts seems to be higher than that of Superhosts.

Neighborhoods: Which Seattle neighborhoods do superhosts tend to have properties in?

In [ ]:
query = """
    SELECT
        DISTINCT l.neighbourhood_cleansed
    FROM
        listing AS l
    JOIN
        host AS h ON l.host_id = h.host_id
    WHERE
        h.host_is_superhost = 1
        AND l.city = 'Seattle'
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Seattle Neighborhoods with Superhost Properties:")
for row in results:
    neighborhood = row[0]
    print(neighborhood)


Seattle Neighborhoods with Superhost Properties:
West Queen Anne
Adams
West Woodland
East Queen Anne
Wallingford
North Queen Anne
Green Lake
Mann
Madrona
University District
Harrison/Denny-Blaine
Minor
Leschi
Atlantic
Pike-Market
Eastlake
South Lake Union
Lawton Park
Briarcliff
Belltown
Central Business District
First Hill
Pioneer Square
Gatewood
Arbor Heights
Alki
North Admiral
Crown Hill
Fairmount Park
Genesee
Interbay
Mid-Beacon Hill
Greenwood
Holly Park
Fauntleroy
North Beacon Hill
Brighton
South Delridge
View Ridge
Dunlap
Rainier Beach
Columbia City
Mount Baker
Seward Park
North Delridge
Maple Leaf
Ravenna
Riverview
Portage Bay
Bryant
Montlake
Broadway
Loyal Heights
Victory Heights
Matthews Beach
Whittier Heights
Meadowbrook
Olympic Hills
Roosevelt
Lower Queen Anne
North Beach/Blue Ridge
Cedar Park
Bitter Lake
Sunset Hill
Haller Lake
North College Park
Phinney Ridge
Windermere
Laurelhurst
Southeast Magnolia
High Point
Seaview
Georgetown
Highland Park
Stevens
Broadview
Fremont


Bedrooms and bathrooms: Do superhosts or other hosts tend to have properties with more bedrooms and bathrooms?

In [ ]:
query = """
    SELECT
        h.host_is_superhost,
        AVG(l.bedrooms) AS average_bedrooms,
        AVG(l.bathrooms) AS average_bathrooms
    FROM
        host AS h
    JOIN
        listing AS l ON h.host_id = l.host_id
    GROUP BY
        h.host_is_superhost
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
for row in results:
    host_type = "Superhost" if row[0] == 1 else "Other Host"
    average_bedrooms = row[1]
    average_bathrooms = row[2]
    print(f"{host_type}: Average Bedrooms - {average_bedrooms:.2f}, Average Bathrooms - {average_bathrooms:.2f}")


Other Host: Average Bedrooms - 0.50, Average Bathrooms - 1.00
Other Host: Average Bedrooms - 1.32, Average Bathrooms - 1.26
Superhost: Average Bedrooms - 1.26, Average Bathrooms - 1.26


Based on the output, we can observe the following trends:

Superhosts and Other Hosts have the same average number of bedrooms, with both groups having an average of 1.26 bedrooms.

Other Hosts tend to have a slightly higher average number of bathrooms compared to Superhosts. The average number of bathrooms for Other Hosts ranges from 1.00 to 1.26, while for Superhosts, it is consistently 1.26.

Therefore, based on these averages, it seems that Other Hosts tend to have properties with more bathrooms, but the number of bedrooms is similar between Superhosts and Other Hosts.

Profile picture: Are superhosts more or less likely to have profile pictures?

In [ ]:
query = """
    SELECT
        COUNT(DISTINCT h.host_id) AS total_hosts,
        COUNT(DISTINCT CASE WHEN h.host_has_profile_pic = 1 THEN h.host_id END) AS hosts_with_profile_pic,
        COUNT(DISTINCT CASE WHEN h.host_has_profile_pic = 1 AND h.host_is_superhost = 1 THEN h.host_id END) AS superhosts_with_profile_pic
    FROM
        host AS h
"""
cursor.execute(query)
results = cursor.fetchone()


In [ ]:
total_hosts = results[0]
hosts_with_profile_pic = results[1]
superhosts_with_profile_pic = results[2]

likelihood_superhost_with_profile_pic = (superhosts_with_profile_pic / hosts_with_profile_pic) * 100

print(f"Total Hosts: {total_hosts}")
print(f"Hosts with Profile Picture: {hosts_with_profile_pic}")
print(f"Superhosts with Profile Picture: {superhosts_with_profile_pic}")
print(f"Likelihood of Superhosts having Profile Pictures: {likelihood_superhost_with_profile_pic:.2f}%")


Total Hosts: 2751
Hosts with Profile Picture: 2742
Superhosts with Profile Picture: 536
Likelihood of Superhosts having Profile Pictures: 19.55%


Based on the output, we can conclude that superhosts are more likely to have profile pictures compared to other hosts. Approximately 19.55% of hosts with a profile picture are identified as superhosts.

This suggests that superhosts, who have achieved a certain level of recognition and positive feedback, are more inclined to provide a profile picture. Having a profile picture may contribute to establishing trust and credibility among guests.

Seattle Neighborhoods: Are superhosts more likely to have properties in specific Seattle neighborhoods? Does their distribution of properties vary across neighborhoods?

In [ ]:
query = """
    SELECT
        l.neighbourhood_cleansed,
        COUNT(DISTINCT CASE WHEN h.host_is_superhost = 1 THEN l.id END) AS superhost_properties,
        COUNT(DISTINCT l.id) AS total_properties
    FROM
        listing AS l
    JOIN
        host AS h ON l.host_id = h.host_id
    WHERE
        l.city = 'Seattle'
    GROUP BY
        l.neighbourhood_cleansed
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Superhost Property Distribution in Seattle Neighborhoods:")
for row in results:
    neighborhood = row[0]
    superhost_properties = row[1]
    total_properties = row[2]
    print(f"Neighborhood: {neighborhood}, Superhost Properties: {superhost_properties}, Total Properties: {total_properties}")


Superhost Property Distribution in Seattle Neighborhoods:
Neighborhood: Adams, Superhost Properties: 22, Total Properties: 70
Neighborhood: Alki, Superhost Properties: 5, Total Properties: 42
Neighborhood: Arbor Heights, Superhost Properties: 1, Total Properties: 5
Neighborhood: Atlantic, Superhost Properties: 11, Total Properties: 53
Neighborhood: Belltown, Superhost Properties: 42, Total Properties: 234
Neighborhood: Bitter Lake, Superhost Properties: 1, Total Properties: 13
Neighborhood: Briarcliff, Superhost Properties: 3, Total Properties: 14
Neighborhood: Brighton, Superhost Properties: 3, Total Properties: 13
Neighborhood: Broadview, Superhost Properties: 2, Total Properties: 17
Neighborhood: Broadway, Superhost Properties: 59, Total Properties: 396
Neighborhood: Bryant, Superhost Properties: 8, Total Properties: 29
Neighborhood: Cedar Park, Superhost Properties: 1, Total Properties: 8
Neighborhood: Central Business District, Superhost Properties: 21, Total Properties: 103
Neigh

Property Size: Are superhosts more likely to have larger or smaller properties (in terms of bedrooms, bathrooms, or square footage) compared to regular hosts?


In [ ]:
query = """
    SELECT
        CASE
            WHEN h.host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        AVG(l.bedrooms) AS average_bedrooms,
        AVG(l.bathrooms) AS average_bathrooms,
        AVG(l.square_feet) AS average_square_feet
    FROM
        listing AS l
    JOIN
        host AS h ON l.host_id = h.host_id
    GROUP BY
        host_type
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Property Size Comparison: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    average_bedrooms = row[1]
    average_bathrooms = row[2]
    average_square_feet = row[3]
    print(f"Host Type: {host_type}")
    print(f"Average Bedrooms: {average_bedrooms:.2f}")
    print(f"Average Bathrooms: {average_bathrooms:.2f}")
    print(f"Average Square Feet: {average_square_feet:.2f}")
    print()


Property Size Comparison: Superhosts vs. Other Hosts
Host Type: Other Host
Average Bedrooms: 1.32
Average Bathrooms: 1.26
Average Square Feet: 844.40

Host Type: Superhost
Average Bedrooms: 1.26
Average Bathrooms: 1.26
Average Square Feet: 878.59



When considering the property size in terms of bedrooms, bathrooms, and square footage, the analysis suggests that superhosts and other hosts are relatively similar. Superhosts have a slightly lower average number of bedrooms but have slightly larger properties in terms of square footage compared to other hosts.

Reservation Patterns: Do superhosts have a different distribution of reservation durations (e.g., minimum nights, maximum nights) compared to regular hosts?

In [ ]:
query = """
    SELECT
        CASE
            WHEN h.host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        AVG(l.minimum_nights) AS average_minimum_nights,
        AVG(l.maximum_nights) AS average_maximum_nights
    FROM
        listing AS l
    JOIN
        host AS h ON l.host_id = h.host_id
    GROUP BY
        host_type
"""
cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Reservation Patterns: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    average_minimum_nights = row[1]
    average_maximum_nights = row[2]
    print(f"Host Type: {host_type}")
    print(f"Average Minimum Nights: {average_minimum_nights:.2f}")
    print(f"Average Maximum Nights: {average_maximum_nights:.2f}")
    print()


Reservation Patterns: Superhosts vs. Other Hosts
Host Type: Other Host
Average Minimum Nights: 2.42
Average Maximum Nights: 812.44

Host Type: Superhost
Average Minimum Nights: 2.18
Average Maximum Nights: 655.43



When considering reservation patterns and durations, the analysis suggests that other hosts generally have slightly longer minimum stay requirements and significantly longer maximum stay durations compared to superhosts. This implies that other hosts may have more flexibility in accommodating longer-term stays, while superhosts tend to have shorter minimum and maximum stay requirements.

Review Scores: Are there notable differences in review scores (e.g., ratings, accuracy, cleanliness, communication) between superhosts and regular hosts?


In [ ]:
query = """
    SELECT
        CASE
            WHEN h.host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        AVG(l.review_scores_rating) AS average_rating,
        AVG(l.review_scores_accuracy) AS average_accuracy,
        AVG(l.review_scores_cleanliness) AS average_cleanliness,
        AVG(l.review_scores_communication) AS average_communication
    FROM
        listing AS l
    JOIN
        host AS h ON l.host_id = h.host_id
    GROUP BY
        host_type
"""

cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Review Scores: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    average_rating = row[1]
    average_accuracy = row[2]
    average_cleanliness = row[3]
    average_communication = row[4]
    print(f"Host Type: {host_type}")
    print(f"Average Rating: {average_rating:.2f}")
    print(f"Average Accuracy: {average_accuracy:.2f}")
    print(f"Average Cleanliness: {average_cleanliness:.2f}")
    print(f"Average Communication: {average_communication:.2f}")
    print()

Review Scores: Superhosts vs. Other Hosts
Host Type: Other Host
Average Rating: 93.65
Average Accuracy: 9.55
Average Cleanliness: 9.45
Average Communication: 9.76

Host Type: Superhost
Average Rating: 97.40
Average Accuracy: 9.91
Average Cleanliness: 9.89
Average Communication: 9.97



When comparing review scores between superhosts and other hosts, the analysis reveals that superhosts tend to receive higher ratings, have higher accuracy and cleanliness scores, and are perceived to have better communication with guests. These findings indicate that superhosts generally excel in providing a positive guest experience, which aligns with their selection as superhosts based on their high ratings, reservation quantity, and low cancellation rates.

Response Time and Rates: Do superhosts have faster response times and higher response rates to inquiries compared to regular hosts?

In [ ]:
query = """
    SELECT
        CASE
            WHEN h.host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        AVG(CASE WHEN h.host_response_time = 'within an hour' THEN 1 ELSE 0 END) AS within_an_hour_rate,
        AVG(CASE WHEN h.host_response_time = 'within a few hours' THEN 1 ELSE 0 END) AS within_a_few_hours_rate,
        AVG(CASE WHEN h.host_response_time = 'within a day' THEN 1 ELSE 0 END) AS within_a_day_rate,
        AVG(CASE WHEN h.host_response_time = 'a few days or more' THEN 1 ELSE 0 END) AS a_few_days_or_more_rate,
        AVG(h.host_response_rate) AS average_response_rate
    FROM
        host AS h
    GROUP BY
        host_type
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Response Time and Rates: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    within_an_hour_rate = row[1] * 100
    within_a_few_hours_rate = row[2] * 100
    within_a_day_rate = row[3] * 100
    a_few_days_or_more_rate = row[4] * 100
    average_response_rate = row[5] * 100
    print(f"Host Type: {host_type}")
    print(f"Within an Hour Rate: {within_an_hour_rate:.2f}%")
    print(f"Within a Few Hours Rate: {within_a_few_hours_rate:.2f}%")
    print(f"Within a Day Rate: {within_a_day_rate:.2f}%")
    print(f"A Few Days or More Rate: {a_few_days_or_more_rate:.2f}%")
    print(f"Average Response Rate: {average_response_rate:.2f}%")
    print()

Response Time and Rates: Superhosts vs. Other Hosts
Host Type: Other Host
Within an Hour Rate: 35.05%
Within a Few Hours Rate: 23.17%
Within a Day Rate: 19.02%
A Few Days or More Rate: 1.40%
Average Response Rate: 9358.24%

Host Type: Superhost
Within an Hour Rate: 56.42%
Within a Few Hours Rate: 28.68%
Within a Day Rate: 8.94%
A Few Days or More Rate: 0.19%
Average Response Rate: 9808.70%



Overall, the analysis suggests that superhosts have faster response times and higher response rates compared to regular hosts. They are more likely to respond within an hour or a few hours, demonstrating their promptness and attentiveness to guest inquiries. Additionally, superhosts exhibit a higher average response rate, indicating a strong commitment to providing exceptional guest communication. These findings align with the criteria for selecting superhosts, which include high ratings, reservation quantity, and minimal cancellations, and further highlight the superior level of service offered by superhosts.

Host Listings Count: Do superhosts tend to have a larger number of listings compared to regular hosts?

In [ ]:
query = """
    SELECT
        CASE
            WHEN host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        AVG(host_listings_count) AS average_listings_count
    FROM
        host
    GROUP BY
        host_type
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Host Listings Count: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    average_listings_count = row[1]
    print(f"Host Type: {host_type}")
    print(f"Average Listings Count: {average_listings_count:.2f}")
    print()

Host Listings Count: Superhosts vs. Other Hosts
Host Type: Other Host
Average Listings Count: 2.03

Host Type: Superhost
Average Listings Count: 1.54



Superhosts tend to have a smaller average number of listings (1.54) compared to other hosts (2.03). This suggests that regular hosts, who are not classified as superhosts, have a slightly higher tendency to manage a larger number of listings.

The difference in the average listings count indicates that superhosts may focus more on maintaining a smaller number of properties, potentially allowing them to provide more personalized attention and care to each listing. On the other hand, other hosts, who are not superhosts, may have a larger portfolio of listings, which could suggest a different business strategy or approach to property management.

Overall, this analysis suggests that superhosts tend to have a smaller number of listings compared to regular hosts.

Host Identity Verification: Are superhosts more likely to have their identities verified compared to regular hosts

In [ ]:
query = """
    SELECT
        CASE
            WHEN host_is_superhost = 1 THEN 'Superhost'
            ELSE 'Other Host'
        END AS host_type,
        COUNT(*) AS count,
        AVG(CASE WHEN host_identity_verified = 1 THEN 1 ELSE 0 END) AS identity_verified_rate
    FROM
        host
    GROUP BY
        host_type
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Host Identity Verification: Superhosts vs. Other Hosts")
for row in results:
    host_type = row[0]
    count = row[1]
    identity_verified_rate = row[2] * 100
    print(f"Host Type: {host_type}")
    print(f"Total Hosts: {count}")
    print(f"Identity Verified Rate: {identity_verified_rate:.2f}%")
    print()

Host Identity Verification: Superhosts vs. Other Hosts
Host Type: Other Host
Total Hosts: 2214
Identity Verified Rate: 72.04%

Host Type: Superhost
Total Hosts: 537
Identity Verified Rate: 87.90%



Superhosts have a higher likelihood of having their identities verified compared to other hosts. The identity verified rate for superhosts is 87.90%, indicating that a significant majority of superhosts have undergone the identity verification process. On the other hand, the identity verified rate for other hosts is 72.04%, which is relatively lower but still significant.

The higher identity verified rate among superhosts suggests that they prioritize and value the identity verification process. It could be attributed to the fact that superhosts, who have achieved a high rating, a large number of reservations, and minimal cancellations, are more inclined to provide a higher level of transparency and trustworthiness by verifying their identities.

Overall, the analysis indicates that superhosts are more likely to have their identities verified compared to regular hosts. This higher identity verification rate adds an extra layer of credibility and reassurance for guests when booking with superhosts.

Who are the top 10 host based on revenue?

In [ ]:
query = """
    SELECT
        host_id,
        SUM(price * has_availability) AS revenue
    FROM listing
    GROUP BY host_id
    ORDER BY revenue DESC
    LIMIT 10;
"""
cursor.execute(query)
results = cursor.fetchall()

In [ ]:
for row in results:
    host_id, revenue = row
    print(f"Host ID: {host_id}, Revenue: ${revenue}")

Host ID: 8534462, Revenue: $9638.0
Host ID: 430709, Revenue: $7501.0
Host ID: 74305, Revenue: $6206.0
Host ID: 4962900, Revenue: $5730.0
Host ID: 658155, Revenue: $3323.0
Host ID: 3074414, Revenue: $3113.0
Host ID: 754810, Revenue: $2775.0
Host ID: 31148752, Revenue: $2448.0
Host ID: 5325329, Revenue: $2273.0
Host ID: 1243056, Revenue: $1775.0


How is the number of host joined to Airbnb over time?


In [ ]:
query = """
    SELECT strftime('%Y-%m', host_since) AS month_joined, COUNT(DISTINCT host_id) AS num_hosts
    FROM host
    GROUP BY month_joined
    ORDER BY month_joined;
"""
cursor.execute(query)
results = cursor.fetchall()

In [ ]:
for row in results:
    month, num_hosts = row
    print(f"Month: {month}, Number of Hosts: {num_hosts}")

Month: None, Number of Hosts: 2
Month: 2008-11, Number of Hosts: 1
Month: 2009-01, Number of Hosts: 1
Month: 2009-02, Number of Hosts: 1
Month: 2009-03, Number of Hosts: 2
Month: 2009-04, Number of Hosts: 1
Month: 2009-05, Number of Hosts: 1
Month: 2009-06, Number of Hosts: 1
Month: 2009-08, Number of Hosts: 7
Month: 2009-09, Number of Hosts: 2
Month: 2009-10, Number of Hosts: 2
Month: 2009-11, Number of Hosts: 3
Month: 2009-12, Number of Hosts: 4
Month: 2010-01, Number of Hosts: 5
Month: 2010-02, Number of Hosts: 2
Month: 2010-03, Number of Hosts: 5
Month: 2010-04, Number of Hosts: 11
Month: 2010-05, Number of Hosts: 2
Month: 2010-06, Number of Hosts: 1
Month: 2010-07, Number of Hosts: 13
Month: 2010-08, Number of Hosts: 15
Month: 2010-09, Number of Hosts: 12
Month: 2010-10, Number of Hosts: 3
Month: 2010-11, Number of Hosts: 9
Month: 2010-12, Number of Hosts: 6
Month: 2011-01, Number of Hosts: 7
Month: 2011-02, Number of Hosts: 13
Month: 2011-03, Number of Hosts: 11
Month: 2011-04, N

#### Exploring trends in property ratings.

Price of listings: Do high reviews tend to be associated with more expensive or less expensive listings?

In [ ]:
query = """
    SELECT
        CASE
            WHEN review_scores_rating >= 90 THEN 'High Review'
            ELSE 'Low Review'
        END AS review_category,
        AVG(price) AS average_price
    FROM
        listing
    WHERE
        price IS NOT NULL
    GROUP BY
        review_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:

print("Price of Listings: High Reviews vs. Low Reviews")
for row in results:
    review_category = row[0]
    average_price = row[1]
    print(f"Review Category: {review_category}")
    print(f"Average Price: ${average_price:.2f}")
    print()

Price of Listings: High Reviews vs. Low Reviews
Review Category: High Review
Average Price: $128.02

Review Category: Low Review
Average Price: $127.86



There is a minimal difference in the average price of listings between the "High Review" and "Low Review" categories. The average price for listings with high reviews is USD 128.02, while the average price for listings with low reviews is slightly lower at USD 127.86.

This suggests that high reviews are not strongly correlated with significantly higher or lower prices of listings. It indicates that the price of a listing is not solely determined by the review score rating. Other factors, such as location, property amenities, size, and demand, might play a more significant role in determining the price.

Therefore, based on this analysis, there is no clear indication that high reviews tend to be associated with more expensive or less expensive listings. The price of a listing appears to be influenced by various factors beyond the review scores.

Neighborhoods: Which Seattle neighborhoods have the highest reviewed properties?

In [ ]:
query = """
    SELECT
        neighbourhood_cleansed,
        AVG(review_scores_rating) AS average_rating
    FROM
        listing
    WHERE
        review_scores_rating IS NOT NULL
    GROUP BY
        neighbourhood_cleansed
    ORDER BY
        average_rating DESC
    LIMIT 5
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Neighborhoods with Highest Reviewed Properties")
for row in results:
    neighborhood = row[0]
    average_rating = row[1]
    print(f"Neighborhood: {neighborhood}")
    print(f"Average Rating: {average_rating:.2f}")
    print()

Neighborhoods with Highest Reviewed Properties
Neighborhood: Arbor Heights
Average Rating: 98.00

Neighborhood: Riverview
Average Rating: 97.25

Neighborhood: Southeast Magnolia
Average Rating: 97.21

Neighborhood: Madrona
Average Rating: 97.10

Neighborhood: Harrison/Denny-Blaine
Average Rating: 97.08



Bedrooms and bathrooms: Do higher rated properties tend to be larger or smaller?

In [ ]:
query = """
    SELECT
        CASE
            WHEN review_scores_rating >= 90 THEN 'High Rating'
            ELSE 'Low Rating'
        END AS rating_category,
        AVG(bedrooms) AS average_bedrooms,
        AVG(bathrooms) AS average_bathrooms
    FROM
        listing
    WHERE
        bedrooms IS NOT NULL
        AND bathrooms IS NOT NULL
        AND review_scores_rating IS NOT NULL
    GROUP BY
        rating_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Property Size Comparison: High Rating vs. Low Rating")
for row in results:
    rating_category = row[0]
    average_bedrooms = row[1]
    average_bathrooms = row[2]
    print(f"Rating Category: {rating_category}")
    print(f"Average Bedrooms: {average_bedrooms:.2f}")
    print(f"Average Bathrooms: {average_bathrooms:.2f}")
    print()

Property Size Comparison: High Rating vs. Low Rating
Rating Category: High Rating
Average Bedrooms: 1.31
Average Bathrooms: 1.26

Rating Category: Low Rating
Average Bedrooms: 1.29
Average Bathrooms: 1.23



There is a minimal difference in the average number of bedrooms and bathrooms between properties with high ratings and properties with low ratings. For properties in the "High Rating" category, the average number of bedrooms is 1.31 and the average number of bathrooms is 1.26. On the other hand, for properties in the "Low Rating" category, the average number of bedrooms is 1.29 and the average number of bathrooms is 1.23.

This suggests that there is no substantial relationship between the rating of a property and its size in terms of bedrooms and bathrooms. The size of a property, as measured by the number of bedrooms and bathrooms, does not seem to be a significant determining factor in achieving higher ratings.

Other factors such as the overall condition, cleanliness, amenities, location, and the overall experience provided by the host may have a more substantial impact on the ratings of the properties.

Therefore, based on this analysis, there is no clear indication that higher rated properties are consistently larger or smaller in terms of bedrooms and bathrooms. The rating of a property appears to be influenced by a variety of factors beyond its size.

Availability of listings: Do higher rated properties tend to have less availability than lower rated properties?

In [ ]:
query = """
    SELECT
        CASE
            WHEN review_scores_rating >= 90 THEN 'High Rating'
            ELSE 'Low Rating'
        END AS rating_category,
        AVG(has_availability) AS average_availability
    FROM
        listing
    WHERE
        review_scores_rating IS NOT NULL
    GROUP BY
        rating_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Availability of Listings: High Rating vs. Low Rating")
for row in results:
    rating_category = row[0]
    average_availability = row[1]
    print(f"Rating Category: {rating_category}")
    print(f"Average Availability: {average_availability:.2f}")
    print()

Availability of Listings: High Rating vs. Low Rating
Rating Category: High Rating
Average Availability: 1.00

Rating Category: Low Rating
Average Availability: 1.00



There is no noticeable difference in the average availability between properties with high ratings and properties with low ratings. Both categories have an average availability of 1.00.

This suggests that the rating of a property does not have a significant impact on its availability. The availability of a property, as represented by the has_availability column, seems to be consistent regardless of its rating. Higher rated properties are not necessarily more or less available compared to lower rated properties.

Other factors such as the host's scheduling preferences, booking policies, and the overall demand for the property may influence its availability more than its rating. The availability of a property is likely determined by factors beyond its rating, making it an independent aspect of the property listing.

Neighborhood Concentration: Are highly rated properties more concentrated in specific neighborhoods compared to lowly rated properties?

In [ ]:
query = """
    SELECT
        neighbourhood_cleansed,
        AVG(CASE WHEN review_scores_rating >= 90 THEN 1 ELSE 0 END) AS high_rating_concentration,
        AVG(CASE WHEN review_scores_rating < 90 THEN 1 ELSE 0 END) AS low_rating_concentration
    FROM
        listing
    GROUP BY
        neighbourhood_cleansed
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Neighborhood Concentration: Highly Rated Properties vs. Lowly Rated Properties")
for row in results:
    neighborhood = row[0]
    high_rating_concentration = row[1]
    low_rating_concentration = row[2]
    print(f"Neighborhood: {neighborhood}")
    print(f"Highly Rated Property Concentration: {high_rating_concentration:.2%}")
    print(f"Lowly Rated Property Concentration: {low_rating_concentration:.2%}")
    print()

Neighborhood Concentration: Highly Rated Properties vs. Lowly Rated Properties
Neighborhood: Adams
Highly Rated Property Concentration: 77.14%
Lowly Rated Property Concentration: 4.29%

Neighborhood: Alki
Highly Rated Property Concentration: 61.90%
Lowly Rated Property Concentration: 4.76%

Neighborhood: Arbor Heights
Highly Rated Property Concentration: 60.00%
Lowly Rated Property Concentration: 0.00%

Neighborhood: Atlantic
Highly Rated Property Concentration: 64.15%
Lowly Rated Property Concentration: 15.09%

Neighborhood: Belltown
Highly Rated Property Concentration: 72.65%
Lowly Rated Property Concentration: 14.10%

Neighborhood: Bitter Lake
Highly Rated Property Concentration: 69.23%
Lowly Rated Property Concentration: 23.08%

Neighborhood: Briarcliff
Highly Rated Property Concentration: 57.14%
Lowly Rated Property Concentration: 28.57%

Neighborhood: Brighton
Highly Rated Property Concentration: 84.62%
Lowly Rated Property Concentration: 0.00%

Neighborhood: Broadview
Highly Rat

Instant Bookability: Are highly rated properties more likely to be instantly bookable compared to lowly rated properties?

In [ ]:
query = """
    SELECT
        CASE WHEN review_scores_rating >= 90 THEN 'Highly Rated' ELSE 'Lowly Rated' END AS rating_category,
        AVG(CASE WHEN instant_bookable = 't' THEN 1 ELSE 0 END) AS instant_bookability_rate
    FROM
        listing
    GROUP BY
        rating_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Instant Bookability: Highly Rated Properties vs. Lowly Rated Properties")
for row in results:
    rating_category = row[0]
    instant_bookability_rate = row[1]
    print(f"Rating Category: {rating_category}")
    print(f"Instant Bookability Rate: {instant_bookability_rate:.2%}")
    print()

Instant Bookability: Highly Rated Properties vs. Lowly Rated Properties
Rating Category: Highly Rated
Instant Bookability Rate: 0.00%

Rating Category: Lowly Rated
Instant Bookability Rate: 0.00%



We can't understand much from this output

Cancellation Policy: Do highly rated properties have different cancellation policies compared to lowly rated properties?

In [ ]:
query = """
    SELECT
        CASE WHEN review_scores_rating >= 90 THEN 'Highly Rated' ELSE 'Lowly Rated' END AS rating_category,
        cancellation_policy,
        COUNT(*) AS property_count
    FROM
        listing
    GROUP BY
        rating_category,
        cancellation_policy
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Cancellation Policy: Highly Rated Properties vs. Lowly Rated Properties")
for row in results:
    rating_category = row[0]
    cancellation_policy = row[1]
    property_count = row[2]
    print(f"Rating Category: {rating_category}")
    print(f"Cancellation Policy: {cancellation_policy}")
    print(f"Property Count: {property_count}")
    print()

Cancellation Policy: Highly Rated Properties vs. Lowly Rated Properties
Rating Category: Highly Rated
Cancellation Policy: flexible
Property Count: 668

Rating Category: Highly Rated
Cancellation Policy: moderate
Property Count: 1004

Rating Category: Highly Rated
Cancellation Policy: strict
Property Count: 1073

Rating Category: Lowly Rated
Cancellation Policy: flexible
Property Count: 482

Rating Category: Lowly Rated
Cancellation Policy: moderate
Property Count: 247

Rating Category: Lowly Rated
Cancellation Policy: strict
Property Count: 344



Based on this analysis, there appears to be a notable difference in the distribution of cancellation policies between highly rated properties and lowly rated properties. Highly rated properties tend to have a higher proportion of properties with strict cancellation policies, while lowly rated properties have a higher proportion of properties with flexible and moderate cancellation policies. This suggests that cancellation policies may play a role in the overall rating of the property by guests.

Is there any correlation between room price and the review score?


In [ ]:
query = "SELECT price, review_scores_rating \
         FROM listing"


cursor.execute(query)
results = cursor.fetchall()

In [ ]:
room_prices = []
review_scores = []
for row in results:
    room_price, review_score = row
    room_prices.append(room_price)
    review_scores.append(review_score)

print("Room Prices:")
print(room_prices)
print("\nReview Scores:")
print(review_scores)

Room Prices:
[85.0, 150.0, 975.0, 100.0, 450.0, 120.0, 80.0, 60.0, 90.0, 150.0, 95.0, 99.0, 245.0, 165.0, 461.0, 109.0, 66.0, 99.0, 200.0, 700.0, 110.0, 99.0, 450.0, 75.0, 600.0, 450.0, 150.0, 45.0, 300.0, 80.0, 175.0, 325.0, 175.0, 222.0, 159.0, 125.0, 348.0, 45.0, 150.0, 80.0, 110.0, 148.0, 350.0, 150.0, 150.0, 100.0, 349.0, 110.0, 160.0, 130.0, 99.0, 100.0, 137.0, 20.0, 150.0, 145.0, 80.0, 95.0, 400.0, 85.0, 70.0, 90.0, 80.0, 60.0, 175.0, 170.0, 85.0, 465.0, 175.0, 50.0, 60.0, 59.0, 47.0, 145.0, 70.0, 55.0, 95.0, 125.0, 65.0, 25.0, 40.0, 149.0, 125.0, 55.0, 129.0, 100.0, 150.0, 50.0, 99.0, 150.0, 105.0, 218.0, 80.0, 126.0, 115.0, 225.0, 89.0, 225.0, 50.0, 134.0, 75.0, 65.0, 375.0, 97.0, 145.0, 129.0, 197.0, 120.0, 200.0, 135.0, 70.0, 150.0, 200.0, 60.0, 180.0, 69.0, 90.0, 80.0, 65.0, 150.0, 195.0, 149.0, 224.0, 115.0, 55.0, 338.0, 79.0, 80.0, 85.0, 80.0, 49.0, 100.0, 79.0, 75.0, 57.0, 99.0, 55.0, 96.0, 99.0, 79.0, 115.0, 350.0, 295.0, 70.0, 53.0, 110.0, 225.0, 35.0, 55.0, 45.0, 79.0

What is the top and bottom 10 property type based on average price and review score?

In [ ]:
# Retrieve the top 10 property types based on average price
query_top_price = '''
SELECT property_type, AVG(price) AS average_price
FROM listing
GROUP BY property_type
ORDER BY average_price DESC
LIMIT 10
'''

results_top_price = cursor.execute(query_top_price).fetchall()

# Retrieve the bottom 10 property types based on average price
query_bottom_price = '''
SELECT property_type, AVG(price) AS average_price
FROM listing
GROUP BY property_type
ORDER BY average_price ASC
LIMIT 10
'''
results_bottom_price = cursor.execute(query_bottom_price).fetchall()


In [ ]:
# Print the top and bottom 10 property types based on average price
print("Top 10 Property Types (Based on Average Price):")
for row in results_top_price:
    property_type, average_price = row
    print(f"{property_type}: {average_price}")

print("\nBottom 10 Property Types (Based on Average Price):")
for row in results_bottom_price:
    property_type, average_price = row
    print(f"{property_type}: {average_price}")


Top 10 Property Types (Based on Average Price):
Boat: 282.375
Condominium: 151.1098901098901
Loft: 135.7
House: 132.35141373341028
Townhouse: 129.45762711864407
Apartment: 122.93266978922716
Camper/RV: 120.46153846153847
None: 120.0
Bungalow: 115.61538461538461
Bed & Breakfast: 112.05405405405405

Bottom 10 Property Types (Based on Average Price):
Dorm: 39.5
Tent: 54.0
Other: 94.68181818181819
Chalet: 98.5
Yurt: 100.0
Treehouse: 101.0
Cabin: 104.66666666666667
Bed & Breakfast: 112.05405405405405
Bungalow: 115.61538461538461
None: 120.0


In [ ]:
# Retrieve the top 10 property types based on average review score
query_top_score = '''
SELECT property_type, AVG(review_scores_rating) AS average_score
FROM listing
GROUP BY property_type
ORDER BY average_score DESC
LIMIT 10
'''
results_top_score = cursor.execute(query_top_score).fetchall()

# Retrieve the bottom 10 property types based on average review score
query_bottom_score = '''
SELECT property_type, AVG(review_scores_rating) AS average_score
FROM listing
GROUP BY property_type
ORDER BY average_score ASC
LIMIT 10
'''
results_bottom_score = cursor.execute(query_bottom_score).fetchall()


In [ ]:
# Print the top and bottom 10 property types based on average review score
print("\nTop 10 Property Types (Based on Average Review Score):")
for row in results_top_score:
    property_type, average_score = row
    print(f"{property_type}: {average_score}")

print("\nBottom 10 Property Types (Based on Average Review Score):")
for row in results_bottom_score:
    property_type, average_score = row
    print(f"{property_type}: {average_score}")


Top 10 Property Types (Based on Average Review Score):
Yurt: 100.0
Bungalow: 97.66666666666667
Townhouse: 96.625
Other: 96.26315789473684
Cabin: 96.23809523809524
Treehouse: 96.0
Bed & Breakfast: 95.73076923076923
Camper/RV: 95.7
Boat: 95.6
Loft: 95.45454545454545

Bottom 10 Property Types (Based on Average Review Score):
None: None
Chalet: 83.5
Dorm: 90.0
Apartment: 93.97759103641457
Tent: 94.5
House: 94.83356164383562
Condominium: 95.31884057971014
Loft: 95.45454545454545
Boat: 95.6
Camper/RV: 95.7


Property types like Boat, Condominium, Loft, House, and Townhouse tend to have higher average prices, while property types like Dorm, Tent, Other, and Chalet tend to have lower average prices.

Property types like Yurt, Bungalow, and Townhouse receive higher average review scores, indicating that they are generally well-regarded by guests. On the other hand, property types like Chalet, and Dorm receive lower average review scores.

These findings suggest that price does not always correlate with review scores. While some higher-priced property types may receive high review scores, there are also lower-priced options that receive positive reviews. It highlights the importance of considering factors beyond just price when assessing the quality and value of a property.

#### Examining property availability across a variety of metrics.

Security deposit: Do properties that require guests to leave security deposits tend to have more or less availability than properties that do not have that requirement?

In [ ]:
query = """
    SELECT
        CASE WHEN security_deposit > 0 THEN 'With Security Deposit' ELSE 'No Security Deposit' END AS deposit_category,
        AVG(has_availability) AS average_availability
    FROM
        listing
    GROUP BY
        deposit_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Availability: Properties with Security Deposit vs. Properties without Security Deposit")
for row in results:
    deposit_category = row[0]
    average_availability = row[1]
    print(f"Deposit Category: {deposit_category}")
    print(f"Average Availability: {average_availability}")
    print()

Availability: Properties with Security Deposit vs. Properties without Security Deposit
Deposit Category: No Security Deposit
Average Availability: 1.0

Deposit Category: With Security Deposit
Average Availability: 1.0



Based on this analysis, there does not appear to be a significant difference in availability between properties with and without security deposits. Both categories have an average availability of 1.0, indicating that they are equally available for booking.

Requires license: Do properties that require guests to show their IDs tend to have more or less availability than properties that do not have that requirement?

In [ ]:
query = """
    SELECT
        CASE WHEN requires_license = 't' THEN 'Requires License' ELSE 'No License Requirement' END AS license_category,
        AVG(has_availability) AS average_availability
    FROM
        listing
    GROUP BY
        license_category
"""

cursor.execute(query)
results = cursor.fetchall()

In [ ]:
print("Availability: Properties with License Requirement vs. Properties without License Requirement")
for row in results:
    license_category = row[0]
    average_availability = row[1]
    print(f"License Category: {license_category}")
    print(f"Average Availability: {average_availability}")
    print()

Availability: Properties with License Requirement vs. Properties without License Requirement
License Category: No License Requirement
Average Availability: 1.0



Based on this analysis, it appears that properties without a license requirement have an average availability of 1.0, indicating that they are consistently available for booking. This suggests that requiring guests to show their IDs does not have a noticeable impact on the availability of properties.

Number of reviews: Is having a lot of reviews associated with properties that tend to not have much availability? Or with properties that have a lot of availability?

In [ ]:
query = """
    SELECT
        CASE
            WHEN number_of_reviews > 0 THEN 'Has Reviews'
            ELSE 'No Reviews'
        END AS review_category,
        AVG(has_availability) AS average_availability
    FROM
        listing
    GROUP BY
        review_category
"""

cursor.execute(query)
results = cursor.fetchall()


In [ ]:
print("Availability: Properties with Reviews vs. Properties without Reviews")
for row in results:
    review_category = row[0]
    average_availability = row[1]
    print(f"Review Category: {review_category}")
    print(f"Average Availability: {average_availability}")
    print()

Availability: Properties with Reviews vs. Properties without Reviews
Review Category: Has Reviews
Average Availability: 1.0

Review Category: No Reviews
Average Availability: 1.0



Both categories, properties with reviews and properties without reviews, have an average availability of 1.0. This indicates that, on average, both types of properties are consistently available for booking throughout the specified time period.

Based on this analysis, there does not appear to be a significant association between the number of reviews a property has and its availability. Properties with and without reviews have similar average availability, suggesting that the number of reviews does not have a noticeable impact on the availability of properties.

Price: Do cheaper properties tend to have more or less availability than more expensive properties?

In [ ]:
query = """
SELECT
    CASE
        WHEN price < 100 THEN 'Cheap'
        WHEN price >= 100 AND price <= 200 THEN 'Moderate'
        WHEN price > 200 THEN 'Expensive'
    END AS Price_Category,
    AVG(has_availability) AS Average_Availability
FROM listing
GROUP BY Price_Category;
"""
cursor.execute(query)
results = cursor.fetchall()

In [ ]:
for row in results:
    price_category, average_availability = row
    print(f"Price Category: {price_category}\nAverage Availability: {average_availability}\n")

Price Category: Cheap
Average Availability: 1.0

Price Category: Expensive
Average Availability: 1.0

Price Category: Moderate
Average Availability: 1.0



there doesn't seem to be a significant difference in availability based on price categories. All price categories, including cheap, moderate, and expensive, have an average availability of 1.0, indicating that properties across different price ranges are equally available for booking.







Property Location: Are properties located in certain neighborhoods or areas more likely to have higher availability compared to properties in other locations?

In [ ]:
query = '''
SELECT
    neighbourhood_cleansed,
    AVG(has_availability ) AS average_availability
FROM listing
GROUP BY neighbourhood_cleansed
ORDER BY average_availability DESC;
'''
cursor.execute(query)
results = cursor.fetchall()

In [ ]:
for row in results:
    neighborhood, average_availability = row
    print(f"Neighborhood: {neighborhood}")
    print(f"Average Availability: {average_availability}\n")

Neighborhood: Yesler Terrace
Average Availability: 1.0

Neighborhood: Windermere
Average Availability: 1.0

Neighborhood: Whittier Heights
Average Availability: 1.0

Neighborhood: Westlake
Average Availability: 1.0

Neighborhood: West Woodland
Average Availability: 1.0

Neighborhood: West Queen Anne
Average Availability: 1.0

Neighborhood: Wedgwood
Average Availability: 1.0

Neighborhood: Wallingford
Average Availability: 1.0

Neighborhood: View Ridge
Average Availability: 1.0

Neighborhood: Victory Heights
Average Availability: 1.0

Neighborhood: University District
Average Availability: 1.0

Neighborhood: Sunset Hill
Average Availability: 1.0

Neighborhood: Stevens
Average Availability: 1.0

Neighborhood: Southeast Magnolia
Average Availability: 1.0

Neighborhood: South Park
Average Availability: 1.0

Neighborhood: South Lake Union
Average Availability: 1.0

Neighborhood: South Delridge
Average Availability: 1.0

Neighborhood: South Beacon Hill
Average Availability: 1.0

Neighborhood

Minimum and Maximum Nights: Do properties with shorter or longer minimum and maximum night stays tend to have higher or lower availability?

In [ ]:
# query = '''
# SELECT
#     minimum_nights,
#     maximum_nights,
#     AVG(has_availability) AS average_availability
# FROM listing
# GROUP BY minimum_nights, maximum_nights
# ORDER BY average_availability DESC;
# '''
# cursor.execute(query)
# results = cursor.fetchall()


In [ ]:
# for row in results:
#     minimum_nights, maximum_nights, average_availability = row
#     print(f"Minimum Nights: {minimum_nights}")
#     print(f"Maximum Nights: {maximum_nights}")
#     print(f"Average Availability: {average_availability}\n")

Minimum Nights: 1
Maximum Nights: 1
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 2
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 3
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 4
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 5
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 6
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 7
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 8
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 9
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 10
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 12
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 13
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 14
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 15
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 16
Average Availability: 1.0

Minimum Nights: 1
Maximum Nights: 20
Average Avai

Property Size: Do larger or smaller properties tend to have higher availability?

In [ ]:
query = '''
SELECT
    size_category,
    AVG(has_availability) AS average_availability
FROM (
    SELECT
        CASE
            WHEN bedrooms + bathrooms <= 2 THEN 'Small'
            WHEN bedrooms + bathrooms > 2 AND bedrooms + bathrooms <= 4 THEN 'Medium'
            WHEN bedrooms + bathrooms > 4 THEN 'Large'
        END AS size_category,
        has_availability
    FROM listing
) AS subquery
GROUP BY size_category;
'''
cursor.execute(query)
results = cursor.fetchall()

In [ ]:
for row in results:
    size_category, average_availability = row
    print(f"Size Category: {size_category}")
    print(f"Average Availability: {average_availability}\n")


Size Category: None
Average Availability: 1.0

Size Category: Large
Average Availability: 1.0

Size Category: Medium
Average Availability: 1.0

Size Category: Small
Average Availability: 1.0



In [ ]:
conn.commit()
conn.close()